stdlib/LinearAlgebra/src/adjtrans.jl

```diff
@@ -267,10 +267,23 @@ Broadcast.broadcast_preserving_zero_d(f, tvs::Union{Number,TransposeAbsVec}...)
 
 ## multiplication *
 
+function _dot_nonrecursive(u, v)
+    lu = length(u)
+    if lu != length(v)
+        throw(DimensionMismatch("first array has length $(lu) which does not match the length of the second, $(length(v))."))
+    end
+    if lu == 0
+        zero(eltype(u)) * zero(eltype(v))
+    else
+        sum(uu*vv for (uu, vv) in zip(u, v))
+    end
+end
+
 # Adjoint/Transpose-vector * vector
-*(u::AdjointAbsVec{T}, v::AbstractVector{T}) where {T<:Number} = dot(u.parent, v)
+*(u::AdjointAbsVec{<:Number}, v::AbstractVector{<:Number}) = dot(u.parent, v)
 *(u::TransposeAbsVec{T}, v::AbstractVector{T}) where {T<:Real} = dot(u.parent, v)
```

Comment:
Thanks for adding that change in. Any reason not to do the equivalent thing in 284 as well?

Reply:
It's causing method ambiguities that I'm not in the mood now of resolving. I've reverted that. I think the `*(::TransVector,::AbstractVector)` is handled somewhere in `matmul.jl`.